third_party/nvidia/lib/Dialect/NVWS/Transforms/Utilities.cpp

```diff
@@ -0,0 +1,50 @@
+#include "Utilities.h"
+#include "triton/Dialect/TritonNvidiaGPU/IR/Dialect.h"
+
+using namespace mlir::triton;
+using namespace mlir::triton::gpu;
+using namespace mlir::triton::nvidia_gpu;
+
+namespace mlir {
+namespace triton {
+namespace nvws {
+
+MemDescType getDataMemDescType(MemDescType memDescType, bool mutableMemory) {
```

Comment:
this is the same as `getBufferViewType` in `PipeliningUtility.h`

Reply:
I cannot find that method, but would be very nice if it exists.

Reply:
`replaceUsesAndPropagateType` in `include/triton/Dialect/TritonGPU/Transforms/Utility.h`